<!DOCTYPE html>
<html lang="en"><head>
<script src="test_2_082025_files/libs/clipboard/clipboard.min.js"></script>
<script src="test_2_082025_files/libs/quarto-html/tabby.min.js"></script>
<script src="test_2_082025_files/libs/quarto-html/popper.min.js"></script>
<script src="test_2_082025_files/libs/quarto-html/tippy.umd.min.js"></script>
<link href="test_2_082025_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="test_2_082025_files/libs/quarto-html/light-border.css" rel="stylesheet">
<link href="test_2_082025_files/libs/quarto-html/quarto-syntax-highlighting-ea385d0e468b0dd5ea5bf0780b1290d9.css" rel="stylesheet" id="quarto-text-highlighting-styles"><meta charset="utf-8">
  <meta name="generator" content="quarto-1.7.33">

  <meta name="author" content=" Lesley Chapman Hannah, Ph.D., M.S. College of Graduate Studies Northeast Ohio Medical University">
  <title>From Raw Data to Insight: Structuring Complexity</title>
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, minimal-ui">
  <link rel="stylesheet" href="test_2_082025_files/libs/revealjs/dist/reset.css">
  <link rel="stylesheet" href="test_2_082025_files/libs/revealjs/dist/reveal.css">
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    div.columns{display: flex; gap: min(4vw, 1.5em);}
    div.column{flex: auto; overflow-x: auto;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    ul.task-list li input[type="checkbox"] {
      width: 0.8em;
      margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
      vertical-align: middle;
    }
  </style>
  <link rel="stylesheet" href="test_2_082025_files/libs/revealjs/dist/theme/quarto-f563837468303362081e247dddd440d0.css">
  <link rel="stylesheet" href="style.css">
  <link href="test_2_082025_files/libs/revealjs/plugin/quarto-line-highlight/line-highlight.css" rel="stylesheet">
  <link href="test_2_082025_files/libs/revealjs/plugin/reveal-menu/menu.css" rel="stylesheet">
  <link href="test_2_082025_files/libs/revealjs/plugin/reveal-menu/quarto-menu.css" rel="stylesheet">
  <link href="test_2_082025_files/libs/revealjs/plugin/reveal-chalkboard/font-awesome/css/all.css" rel="stylesheet">
  <link href="test_2_082025_files/libs/revealjs/plugin/reveal-chalkboard/style.css" rel="stylesheet">
  <link href="test_2_082025_files/libs/revealjs/plugin/quarto-support/footer.css" rel="stylesheet">
  <style type="text/css">
    .reveal div.sourceCode {
      margin: 0;
      overflow: auto;
    }
    .reveal div.hanging-indent {
      margin-left: 1em;
      text-indent: -1em;
    }
    .reveal .slide:not(.center) {
      height: 100%;
    }
    .reveal .slide.scrollable {
      overflow-y: auto;
    }
    .reveal .footnotes {
      height: 100%;
      overflow-y: auto;
    }
    .reveal .slide .absolute {
      position: absolute;
      display: block;
    }
    .reveal .footnotes ol {
      counter-reset: ol;
      list-style-type: none; 
      margin-left: 0;
    }
    .reveal .footnotes ol li:before {
      counter-increment: ol;
      content: counter(ol) ". "; 
    }
    .reveal .footnotes ol li > p:first-child {
      display: inline-block;
    }
    .reveal .slide ul,
    .reveal .slide ol {
      margin-bottom: 0.5em;
    }
    .reveal .slide ul li,
    .reveal .slide ol li {
      margin-top: 0.4em;
      margin-bottom: 0.2em;
    }
    .reveal .slide ul[role="tablist"] li {
      margin-bottom: 0;
    }
    .reveal .slide ul li > *:first-child,
    .reveal .slide ol li > *:first-child {
      margin-block-start: 0;
    }
    .reveal .slide ul li > *:last-child,
    .reveal .slide ol li > *:last-child {
      margin-block-end: 0;
    }
    .reveal .slide .columns:nth-child(3) {
      margin-block-start: 0.8em;
    }
    .reveal blockquote {
      box-shadow: none;
    }
    .reveal .tippy-content>* {
      margin-top: 0.2em;
      margin-bottom: 0.7em;
    }
    .reveal .tippy-content>*:last-child {
      margin-bottom: 0.2em;
    }
    .reveal .slide > img.stretch.quarto-figure-center,
    .reveal .slide > img.r-stretch.quarto-figure-center {
      display: block;
      margin-left: auto;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-left,
    .reveal .slide > img.r-stretch.quarto-figure-left  {
      display: block;
      margin-left: 0;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-right,
    .reveal .slide > img.r-stretch.quarto-figure-right  {
      display: block;
      margin-left: auto;
      margin-right: 0; 
    }
  </style>
</head>
<body class="quarto-light">
  <div class="reveal">
    <div class="slides">

<section id="title-slide" class="quarto-title-block center">
  <h2 style="color: #2A76DD;" class="title">From Raw Data to Insight: Structuring Complexity</h2>

<div class="quarto-title-authors">
<div class="quarto-title-author">
<div style="font-size:20px;" class="quarto-title-author-name">
<br><br><span style="font-size:25px;"><strong>Lesley Chapman Hannah, Ph.D., M.S.</strong></span><br>College of Graduate Studies<br>Northeast Ohio Medical University 
</div>
</div>
</div>

</section>
<section id="lecture-outline" class="slide level2 smaller">
<h2><span style="font-size:80%">Lecture Outline</span></h2>
<ul>
<li><p>Review: turn high dimensional data into insight</p>
<ul>
<li>Understand how complex, multi-variable analyses—such as gene expression, proteomic, or variant data—are distilled into interpretable biological patterns through plots</li>
</ul></li>
<li><p>Mapping relationships and biological connectivity through biological networks</p>
<ul>
<li>Learn how network and pathway visualizations uncover the relational geometry of biological systems</li>
</ul></li>
<li><p>Aggregating evidence from high-dimensional data into pathways</p>
<ul>
<li>Understand how pathway enrichment plots connect quantitative outcomes to biological mechanisms</li>
</ul></li>
</ul>
<div class="footer">

</div>
</section>
<section id="recap-high-dimensional-hd-data-analysis" class="slide level2 smaller">
<h2><span style="font-size:80%">Recap: High-dimensional [HD] Data Analysis</span></h2>
<p>Sources of HD data:</p>
<ul>
<li><p>Arises from modern -omics, imaging, EHR, and wearable streams</p></li>
<li><p>Millions of features per patient, making dimensionality reduction and organization essential for interpretation</p></li>
<li><p>Transforms data complexity into perceivable structure—turns thousands of features into a spatial or conceptual map of biological systems</p></li>
</ul>
<p>HD Analysis Strategy</p>
<ul>
<li><p>Dimensionality reduction reveals global structure—clusters, gradients, trajectories</p></li>
<li><p>Reveals where patterns live but does not explain biological mechanism</p></li>
</ul>
<p>HD visualization reveals global structures:</p>
<ul>
<li><p>HD data visualization reveals connectivity between biologically features forming meaningful connections rarely driven by single genes or measurements</p></li>
<li><p>Connectivity-based visuals that make these relationships visible: gene co-expression networks, functional modules, pathway cross-talk maps</p></li>
</ul>
<div class="footer">

</div>
</section>
<section id="connecting-statistical-output-to-biological-interpretation-in-high-dimensional-data-analysis" class="slide level2 smaller">
<h2><span style="font-size:80%">Connecting Statistical Output to Biological Interpretation in High-Dimensional Data Analysis</span></h2>
<p>High-dimensional data analysis can produce thousands of statistical results: differential expression tests, feature correlations, variant associations numbers to interpret directly</p>
<p>Visualization transforms this statistical matrix into visual reasoning using the following two approaches:</p>
<ul>
<li><p>Networks <span class="math inline">\(\rightarrow\)</span> How do features relate?</p></li>
<li><p>Pathways <span class="math inline">\(\rightarrow\)</span> What biological programs explain this pattern?</p></li>
</ul>
<div class="footer">

</div>
</section>
<section id="connecting-statistical-output-to-biological-interpretation-in-high-dimensional-data-analysis-1" class="slide level2 smaller">
<h2><span style="font-size:80%">Connecting Statistical Output to Biological Interpretation in High-Dimensional Data Analysis</span></h2>
<p>High-dimensional data analysis can produce thousands of statistical results: differential expression tests, feature correlations, variant associations numbers to interpret directly</p>
<p>Visualization transforms this statistical matrix into visual reasoning using the following approaches:</p>
<ul>
<li><p>Networks <span class="math inline">\(\rightarrow\)</span> How do features relate?</p>
<ul>
<li><p>In high-dimensional space, features (genes, proteins, variants) form co-expression or co-regulation modules</p></li>
<li><p>Network visualizations reduce dimensionality by mapping correlations into a perceivable structure — revealing functional neighborhoods within the data</p></li>
<li><p>Example: Thousands of genes/features <span class="math inline">\(\rightarrow\)</span> network modules reduce complexity by grouping correlated features into “neighbourhoods”</p></li>
</ul></li>
</ul>
<div class="footer">

</div>
</section>
<section id="connecting-statistical-output-to-biological-interpretation-in-high-dimensional-data-analysis-2" class="slide level2 smaller">
<h2><span style="font-size:80%">Connecting Statistical Output to Biological Interpretation in High-Dimensional Data Analysis</span></h2>
<div class="columns">
<div class="column" style="width:50%;">

</div><div class="column" style="width:50%;">
<p><img data-src="images/figure_1.png" class="border border-thick"></p>
</div><p><span style="font-size:50%">Rolland, T. et al.&nbsp;Cell 159, 1212–1226 (2014)</span></p>
</div>
<div class="footer">

</div>
</section>
<section id="networks-and-modules-seeing-connectivity" class="slide level2 smaller">
<h2><span style="font-size:80%">Networks and Modules: Seeing Connectivity</span></h2>
<ul>
<li><p>In HD biology, relationships among features are often more important than the features themselves</p></li>
<li><p>Gene co-expression networks reveal groups of genes that rise/fall together—suggesting shared regulatory control</p></li>
<li><p>Protein–protein interaction (PPI) graphs identify modules that cooperate in signaling, metabolism, or immune activation</p></li>
<li><p>Pathway maps collapse thousands of interacting genes into interpretable sub-networks.</p></li>
<li><p>Visual encodings provide mechanistic meaning:</p>
<ul>
<li><p>Color <span class="math inline">\(\rightarrow\)</span> module identity</p></li>
<li><p>Edge transparency <span class="math inline">\(\rightarrow\)</span> strength of correlation/interaction</p></li>
<li><p>Node size <span class="math inline">\(\rightarrow\)</span> importance (centrality, hubness)</p></li>
</ul></li>
</ul>
<div class="footer">

</div>
</section>
<section id="data-visualization-for-gene-co-expression-networks" class="slide level2 smaller">
<h2><span style="font-size:80%">Data Visualization for Gene Co-expression Networks</span></h2>
<p><strong>Biological Question</strong>: How do genes within the same biological process—such as cytokine signaling, interferon response, or metabolic regulation—co-vary across samples, and which genes serve as key connectors within these processes?</p>
<p><strong>Data [simulated]</strong>:</p>
<ul>
<li><p>40 samples</p></li>
<li><p>15 genes</p></li>
</ul>
<p><strong>Chart type</strong>:</p>
<ul>
<li><p>network graph represents relationships (edges) between entities (nodes)</p></li>
<li><p>Nodes <span class="math inline">\(\rightarrow\)</span> genes</p></li>
<li><p>Edges <span class="math inline">\(\rightarrow\)</span> strong pairwise correlations in gene expression</p></li>
</ul>
<p><strong>Goal(s)</strong>:</p>
<ul>
<li><p>Reveal clusters or modules of genes that function together</p></li>
<li><p>Highlight hub genes, which may regulate or integrate biological pathways</p></li>
</ul>
<div class="footer">

</div>
<style>
.tiny-table table { font-size: 0.8em; }        /* adjust as needed (0.8–0.9em) */
.tiny-table figcaption, .tiny-table .table-caption { font-size: 0.75em; }
</style>
</section>
<section id="data-visualization-for-gene-co-expression-networks-1" class="slide level2 smaller">
<h2><span style="font-size:60%">Data Visualization for Gene Co-expression Networks</span></h2>
<div class="columns">
<div class="column small">
<p><strong>Plot Interpretation</strong>:</p>
<ul>
<li><p><strong>Each point</strong> (node) represents a gene</p>
<ul>
<li>Point positions are derived from pairwise gene–gene correlations, so genes that behave similarly across patients or conditions appear closer together</li>
</ul></li>
<li><p><strong>Solid grey edges</strong> link genes whose expression patterns are strongly correlated (|r| &gt; threshold)</p>
<ul>
<li>These edges show co-expression relationships, mapping the local structure of molecular coordination</li>
</ul></li>
<li><p><strong>Color encodes biological modules</strong>: Cytokine, Interferon, and Metabolic — allowing us to see whether genes cluster by known functional groupings</p></li>
<li><p><strong>Node size</strong> represents hubness (degree)</p>
<ul>
<li>Larger nodes are genes that co-vary with many others and may act as key regulators or integrators of biological signals</li>
</ul></li>
</ul>
</div><div class="column">
<div class="cell">
<div class="cell-output-display">
<div>
<figure>
<p><img data-src="test_2_082025_files/figure-revealjs/unnamed-chunk-1-1.png" width="768"></p>
</figure>
</div>
</div>
</div>
</div></div>
<style>
.tiny-table table { font-size: 0.8em; }        /* adjust as needed (0.8–0.9em) */
.tiny-table figcaption, .tiny-table .table-caption { font-size: 0.75em; }
</style>
<style>
.tiny-table table { font-size: 0.8em; }        /* adjust as needed (0.8–0.9em) */
.tiny-table figcaption, .tiny-table .table-caption { font-size: 0.75em; }
</style>
</section>
<section id="data-visualization-for-gene-co-expression-networks-2" class="slide level2 smaller">
<h2><span style="font-size:60%">Data Visualization for Gene Co-expression Networks</span></h2>
<div class="columns">
<div class="column small">
<p><strong>Plot Interpretation</strong>:</p>
<ul>
<li><strong>Green cluster - cytokine cluster</strong>:</li>
</ul>
<p>Genes likely to coordinate inflammatory communication between cells</p>
<ul>
<li><strong>Orange cluster - interferon cluster</strong>:</li>
</ul>
<p>Genes co-activated as part of the antiviral defense response, typically regulated by interferon pathways</p>
<ul>
<li><strong>Blue cluster - metabolic cluster</strong>:</li>
</ul>
<p>Genes whose expression reflects shared metabolic or energy-regulation processes, providing the cellular resources needed to sustain immune activity</p>
</div><div class="column">
<div class="cell">
<div class="cell-output-display">
<div>
<figure>
<p><img data-src="test_2_082025_files/figure-revealjs/unnamed-chunk-2-1.png" width="768"></p>
</figure>
</div>
</div>
</div>
</div></div>
<style>
.tiny-table table { font-size: 0.8em; }        /* adjust as needed (0.8–0.9em) */
.tiny-table figcaption, .tiny-table .table-caption { font-size: 0.75em; }
</style>
</section>
<section id="connecting-statistical-output-to-biological-interpretation-in-high-dimensional-data-analysis-3" class="slide level2 smaller">
<h2><span style="font-size:80%">Connecting Statistical Output to Biological Interpretation in High-Dimensional Data Analysis</span></h2>
<p>High-dimensional data analysis can produce thousands of statistical results: differential expression tests, feature correlations, variant associations numbers to interpret directly</p>
<p>Visualization transforms this statistical matrix into visual reasoning using the following approaches:</p>
<ul>
<li><p>Pathways <span class="math inline">\(\rightarrow\)</span> What biological programs explain this pattern?</p>
<ul>
<li><p>Once subsections of data are visible, mapping them to pathways or ontologies connects the statistical pattern to biological mechanism — turning feature clusters into interpretable processes</p></li>
<li><p>Example: module corresponds to a T-cell activation pathway / metabolic shift</p></li>
</ul></li>
</ul>
<div class="footer">

</div>
</section>
<section id="pathway-level-analysis-organizing-thousands-of-tests" class="slide level2 smaller">
<h2><span style="font-size:80%">Pathway Level Analysis: Organizing Thousands of Tests</span></h2>
<ul>
<li><p>Organize results into interpretable biological systems</p></li>
<li><p>Data visualization strategies to transform tabular data containing significant genes into visual structures</p></li>
<li><p>Visual aggregation = clarity: grouping results by pathways (GO, KEGG, Reactome)</p></li>
</ul>
<div class="footer">

</div>
</section>
<section id="pathway-level-analysis-organizing-thousands-of-tests-1" class="slide level2 smaller">
<h2><span style="font-size:80%">Pathway Level Analysis: Organizing Thousands of Tests</span></h2>
<ul>
<li><p>After running biomedical analysis - differential expression, epigenetic analysis, proteomics - obtain thousands of significant hits</p></li>
<li><p>Raw tables of p-values offer no biological clarity—HD data must be grouped into interpretable themes</p></li>
<li><p>Pathway level analysis moves from individual gene results (micro-level) to systems-level meaning (macro-level)</p></li>
</ul>
<div class="footer">

</div>
</section>
<section id="pathway-level-analysis-organizing-thousands-of-tests-2" class="slide level2 smaller">
<h2><span style="font-size:80%">Pathway Level Analysis: Organizing Thousands of Tests</span></h2>
<p>Visual aggregation reveals biological programs, not individual genes</p>
<p>HD pathway level visualization types:</p>
<ul>
<li><p><strong>Dot plots</strong> <span class="math inline">\(\rightarrow\)</span> summarize enrichment significance and effect size across pathways</p></li>
<li><p><strong>Enrichment maps</strong> <span class="math inline">\(\rightarrow\)</span> network-style visuals linking related functions (e.g., immune signaling <span class="math inline">\(\leftrightarrow\)</span> cytokine response)</p></li>
<li><p><strong>Network graphs</strong> <span class="math inline">\(\rightarrow\)</span> reveal cross-talk between processes, bridging molecular and clinical insight</p></li>
</ul>
<div class="footer">

</div>
</section>
<section id="pathway-level-analysis-organizing-thousands-of-tests-3" class="slide level2 smaller">
<h2><span style="font-size:80%">Pathway Level Analysis: Organizing Thousands of Tests</span></h2>
<p><strong>Research Question</strong>: Which biological pathways exhibit coordinated shifts in gene expressio, and how does the gene ratio (overlap / pathway size) highlight the relative strength of each pathway’s activation?</p>
<p><strong>Data [simulated]</strong>:</p>
<ul>
<li><p>3,000 genes with expression changes [log-fold changes] and p-values</p></li>
<li><p>Biological processes drawn from KEGG/Reactome/GO</p></li>
</ul>
<div class="footer">

</div>
<style>
.tiny-table table { font-size: 0.8em; }        /* adjust as needed (0.8–0.9em) */
.tiny-table figcaption, .tiny-table .table-caption { font-size: 0.75em; }
</style>
</section>
<section id="pathway-level-analysis-organizing-thousands-of-tests-4" class="slide level2 smaller">
<h2><span style="font-size:60%">Pathway Level Analysis: Organizing Thousands of Tests</span></h2>
<div class="columns">
<div class="column small">
<p><strong>Plot</strong>:</p>
<ul>
<li><p>Pathway enrichment dot plot</p></li>
<li><p>Plot summarizes thousands of statistical tests into a small number of interpretable biological themes</p></li>
</ul>
<p><strong>Plot Interpretation</strong></p>
<ul>
<li><p>Row <span class="math inline">\(\rightarrow\)</span> pathway</p></li>
<li><p>Dot <span class="math inline">\(\rightarrow\)</span> level of pathway enrichment in the data</p></li>
</ul>
</div><div class="column">
<div class="cell">
<div class="cell-output-display">
<div>
<figure>
<p><img data-src="test_2_082025_files/figure-revealjs/unnamed-chunk-3-1.png" width="768"></p>
</figure>
</div>
</div>
</div>
</div></div>
<style>
.tiny-table table { font-size: 0.8em; }        /* adjust as needed (0.8–0.9em) */
.tiny-table figcaption, .tiny-table .table-caption { font-size: 0.75em; }
</style>
<style>
.tiny-table table { font-size: 0.8em; }        /* adjust as needed (0.8–0.9em) */
.tiny-table figcaption, .tiny-table .table-caption { font-size: 0.75em; }
</style>
</section>
<section id="pathway-level-analysis-organizing-thousands-of-tests-5" class="slide level2 smaller">
<h2><span style="font-size:60%">Pathway Level Analysis: Organizing Thousands of Tests</span></h2>
<div class="columns">
<div class="column small">
<p><strong>Plot Interpretation</strong></p>
<ul>
<li><p><strong>X-axis [Gene Ratio]</strong>: fraction of significant genes in that pathway</p></li>
<li><p><strong>Dot size</strong>: number of overlapping genes (pathway “weight”)</p></li>
<li><p><strong>Pathway label (n=…)</strong>: total number of genes in the pathway database</p></li>
<li><p>Pathways such as PI3K-Akt signaling, ER stress response, interferon signaling, and apoptosis show high gene ratios</p></li>
<li><p>Trends suggest possible changes in immune activation, cell survival signaling, or stress adaptation on a cellular level</p></li>
</ul>
</div><div class="column">
<div class="cell">
<div class="cell-output-display">
<div>
<figure>
<p><img data-src="test_2_082025_files/figure-revealjs/unnamed-chunk-4-1.png" width="768"></p>
</figure>
</div>
</div>
</div>
</div></div>
<style>
.tiny-table table { font-size: 0.8em; }        /* adjust as needed (0.8–0.9em) */
.tiny-table figcaption, .tiny-table .table-caption { font-size: 0.75em; }
</style>
</section>
<section id="summary" class="slide level2 smaller">
<h2><span style="font-size:80%">Summary</span></h2>
<ul>
<li><p>network graphs reduce complexity by mapping correlations between genes into clusters</p></li>
<li><p>visuals expose how genes cooperate within broader cellular systems like cytokine, interferon, or metabolic networks</p></li>
<li><p>data can be encoded as:</p>
<ul>
<li><p>Nodes (i.e.: genes)</p></li>
<li><p>Edges (i.e.: correlations)</p></li>
<li><p>Color (i.e.: biological module)</p></li>
<li><p>Size (i.e.: degree of biological impact)</p></li>
</ul></li>
</ul>
<div class="footer">

</div>

</section>
    </div>
  <div class="quarto-auto-generated-content" style="display: none;">
<p><img src="images/Flame.jpg" class="slide-logo"></p>
<div class="footer footer-default">

</div>
</div></div>

  <script>window.backupDefine = window.define; window.define = undefined;</script>
  <script src="test_2_082025_files/libs/revealjs/dist/reveal.js"></script>
  <!-- reveal.js plugins -->
  <script src="test_2_082025_files/libs/revealjs/plugin/quarto-line-highlight/line-highlight.js"></script>
  <script src="test_2_082025_files/libs/revealjs/plugin/pdf-export/pdfexport.js"></script>
  <script src="test_2_082025_files/libs/revealjs/plugin/reveal-menu/menu.js"></script>
  <script src="test_2_082025_files/libs/revealjs/plugin/reveal-menu/quarto-menu.js"></script>
  <script src="test_2_082025_files/libs/revealjs/plugin/reveal-chalkboard/plugin.js"></script>
  <script src="test_2_082025_files/libs/revealjs/plugin/quarto-support/support.js"></script>
  

  <script src="test_2_082025_files/libs/revealjs/plugin/notes/notes.js"></script>
  <script src="test_2_082025_files/libs/revealjs/plugin/search/search.js"></script>
  <script src="test_2_082025_files/libs/revealjs/plugin/zoom/zoom.js"></script>
  <script src="test_2_082025_files/libs/revealjs/plugin/math/math.js"></script>
  <script>window.define = window.backupDefine; window.backupDefine = undefined;</script>

  <script>

      // Full list of configuration options available at:
      // https://revealjs.com/config/
      Reveal.initialize({
'controlsAuto': true,
'previewLinksAuto': true,
'pdfSeparateFragments': false,
'autoAnimateEasing': "ease",
'autoAnimateDuration': 1,
'autoAnimateUnmatched': true,
'jumpToSlide': true,
'menu': {"side":"left","useTextContentForMissingTitles":true,"markers":false,"loadIcons":false,"custom":[{"title":"Tools","icon":"<i class=\"fas fa-gear\"></i>","content":"<ul class=\"slide-menu-items\">\n<li class=\"slide-tool-item active\" data-item=\"0\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.fullscreen(event)\"><kbd>f</kbd> Fullscreen</a></li>\n<li class=\"slide-tool-item\" data-item=\"1\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.speakerMode(event)\"><kbd>s</kbd> Speaker View</a></li>\n<li class=\"slide-tool-item\" data-item=\"2\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.overview(event)\"><kbd>o</kbd> Slide Overview</a></li>\n<li class=\"slide-tool-item\" data-item=\"3\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.togglePdfExport(event)\"><kbd>e</kbd> PDF Export Mode</a></li>\n<li class=\"slide-tool-item\" data-item=\"4\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.toggleScrollView(event)\"><kbd>r</kbd> Scroll View Mode</a></li>\n<li class=\"slide-tool-item\" data-item=\"5\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.toggleChalkboard(event)\"><kbd>b</kbd> Toggle Chalkboard</a></li>\n<li class=\"slide-tool-item\" data-item=\"6\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.toggleNotesCanvas(event)\"><kbd>c</kbd> Toggle Notes Canvas</a></li>\n<li class=\"slide-tool-item\" data-item=\"7\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.downloadDrawings(event)\"><kbd>d</kbd> Download Drawings</a></li>\n<li class=\"slide-tool-item\" data-item=\"8\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.keyboardHelp(event)\"><kbd>?</kbd> Keyboard Help</a></li>\n</ul>"}],"openButton":true},
'chalkboard': {"buttons":false},
'smaller': false,
 
        // Display controls in the bottom right corner
        controls: false,

        // Help the user learn the controls by providing hints, for example by
        // bouncing the down arrow when they first encounter a vertical slide
        controlsTutorial: false,

        // Determines where controls appear, "edges" or "bottom-right"
        controlsLayout: 'edges',

        // Visibility rule for backwards navigation arrows; "faded", "hidden"
        // or "visible"
        controlsBackArrows: 'faded',

        // Display a presentation progress bar
        progress: true,

        // Display the page number of the current slide
        slideNumber: 'c/t',

        // 'all', 'print', or 'speaker'
        showSlideNumber: 'all',

        // Add the current slide number to the URL hash so that reloading the
        // page/copying the URL will return you to the same slide
        hash: true,

        // Start with 1 for the hash rather than 0
        hashOneBasedIndex: false,

        // Flags if we should monitor the hash and change slides accordingly
        respondToHashChanges: true,

        // Push each slide change to the browser history
        history: true,

        // Enable keyboard shortcuts for navigation
        keyboard: true,

        // Enable the slide overview mode
        overview: true,

        // Disables the default reveal.js slide layout (scaling and centering)
        // so that you can use custom CSS layout
        disableLayout: false,

        // Vertical centering of slides
        center: false,

        // Enables touch navigation on devices with touch input
        touch: true,

        // Loop the presentation
        loop: false,

        // Change the presentation direction to be RTL
        rtl: false,

        // see https://revealjs.com/vertical-slides/#navigation-mode
        navigationMode: 'linear',

        // Randomizes the order of slides each time the presentation loads
        shuffle: false,

        // Turns fragments on and off globally
        fragments: true,

        // Flags whether to include the current fragment in the URL,
        // so that reloading brings you to the same fragment position
        fragmentInURL: false,

        // Flags if the presentation is running in an embedded mode,
        // i.e. contained within a limited portion of the screen
        embedded: false,

        // Flags if we should show a help overlay when the questionmark
        // key is pressed
        help: true,

        // Flags if it should be possible to pause the presentation (blackout)
        pause: true,

        // Flags if speaker notes should be visible to all viewers
        showNotes: false,

        // Global override for autoplaying embedded media (null/true/false)
        autoPlayMedia: null,

        // Global override for preloading lazy-loaded iframes (null/true/false)
        preloadIframes: null,

        // Number of milliseconds between automatically proceeding to the
        // next slide, disabled when set to 0, this value can be overwritten
        // by using a data-autoslide attribute on your slides
        autoSlide: 0,

        // Stop auto-sliding after user input
        autoSlideStoppable: true,

        // Use this method for navigation when auto-sliding
        autoSlideMethod: null,

        // Specify the average time in seconds that you think you will spend
        // presenting each slide. This is used to show a pacing timer in the
        // speaker view
        defaultTiming: null,

        // Enable slide navigation via mouse wheel
        mouseWheel: false,

        // The display mode that will be used to show slides
        display: 'block',

        // Hide cursor if inactive
        hideInactiveCursor: true,

        // Time before the cursor is hidden (in ms)
        hideCursorTime: 5000,

        // Opens links in an iframe preview overlay
        previewLinks: false,

        // Transition style (none/fade/slide/convex/concave/zoom)
        transition: 'none',

        // Transition speed (default/fast/slow)
        transitionSpeed: 'default',

        // Transition style for full page slide backgrounds
        // (none/fade/slide/convex/concave/zoom)
        backgroundTransition: 'none',

        // Number of slides away from the current that are visible
        viewDistance: 3,

        // Number of slides away from the current that are visible on mobile
        // devices. It is advisable to set this to a lower number than
        // viewDistance in order to save resources.
        mobileViewDistance: 2,

        // The "normal" size of the presentation, aspect ratio will be preserved
        // when the presentation is scaled to fit different resolutions. Can be
        // specified using percentage units.
        width: 1050,

        height: 700,

        // Factor of the display size that should remain empty around the content
        margin: 0.1,

        math: {
          mathjax: 'https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/MathJax.js',
          config: 'TeX-AMS_HTML-full',
          tex2jax: {
            inlineMath: [['\\(','\\)']],
            displayMath: [['\\[','\\]']],
            balanceBraces: true,
            processEscapes: false,
            processRefs: true,
            processEnvironments: true,
            preview: 'TeX',
            skipTags: ['script','noscript','style','textarea','pre','code'],
            ignoreClass: 'tex2jax_ignore',
            processClass: 'tex2jax_process'
          },
        },

        // reveal.js plugins
        plugins: [QuartoLineHighlight, PdfExport, RevealMenu, RevealChalkboard, QuartoSupport,

          RevealMath,
          RevealNotes,
          RevealSearch,
          RevealZoom
        ]
      });
    </script>
    
    <script>
      // htmlwidgets need to know to resize themselves when slides are shown/hidden.
      // Fire the "slideenter" event (handled by htmlwidgets.js) when the current
      // slide changes (different for each slide format).
      (function () {
        // dispatch for htmlwidgets
        function fireSlideEnter() {
          const event = window.document.createEvent("Event");
          event.initEvent("slideenter", true, true);
          window.document.dispatchEvent(event);
        }

        function fireSlideChanged(previousSlide, currentSlide) {
          fireSlideEnter();

          // dispatch for shiny
          if (window.jQuery) {
            if (previousSlide) {
              window.jQuery(previousSlide).trigger("hidden");
            }
            if (currentSlide) {
              window.jQuery(currentSlide).trigger("shown");
            }
          }
        }

        // hookup for slidy
        if (window.w3c_slidy) {
          window.w3c_slidy.add_observer(function (slide_num) {
            // slide_num starts at position 1
            fireSlideChanged(null, w3c_slidy.slides[slide_num - 1]);
          });
        }

      })();
    </script>

    <script id="quarto-html-after-body" type="application/javascript">
      window.document.addEventListener("DOMContentLoaded", function (event) {
        const tabsets =  window.document.querySelectorAll(".panel-tabset-tabby")
        tabsets.forEach(function(tabset) {
          const tabby = new Tabby('#' + tabset.id);
        });
        const isCodeAnnotation = (el) => {
          for (const clz of el.classList) {
            if (clz.startsWith('code-annotation-')) {                     
              return true;
            }
          }
          return false;
        }
        const onCopySuccess = function(e) {
          // button target
          const button = e.trigger;
          // don't keep focus
          button.blur();
          // flash "checked"
          button.classList.add('code-copy-button-checked');
          var currentTitle = button.getAttribute("title");
          button.setAttribute("title", "Copied!");
          let tooltip;
          if (window.bootstrap) {
            button.setAttribute("data-bs-toggle", "tooltip");
            button.setAttribute("data-bs-placement", "left");
            button.setAttribute("data-bs-title", "Copied!");
            tooltip = new bootstrap.Tooltip(button, 
              { trigger: "manual", 
                customClass: "code-copy-button-tooltip",
                offset: [0, -8]});
            tooltip.show();    
          }
          setTimeout(function() {
            if (tooltip) {
              tooltip.hide();
              button.removeAttribute("data-bs-title");
              button.removeAttribute("data-bs-toggle");
              button.removeAttribute("data-bs-placement");
            }
            button.setAttribute("title", currentTitle);
            button.classList.remove('code-copy-button-checked');
          }, 1000);
          // clear code selection
          e.clearSelection();
        }
        const getTextToCopy = function(trigger) {
            const codeEl = trigger.previousElementSibling.cloneNode(true);
            for (const childEl of codeEl.children) {
              if (isCodeAnnotation(childEl)) {
                childEl.remove();
              }
            }
            return codeEl.innerText;
        }
        const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
          text: getTextToCopy
        });
        clipboard.on('success', onCopySuccess);
        if (window.document.getElementById('quarto-embedded-source-code-modal')) {
          const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
            text: getTextToCopy,
            container: window.document.getElementById('quarto-embedded-source-code-modal')
          });
          clipboardModal.on('success', onCopySuccess);
        }
          var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
          var mailtoRegex = new RegExp(/^mailto:/);
            var filterRegex = new RegExp('/' + window.location.host + '/');
          var isInternal = (href) => {
              return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
          }
          // Inspect non-navigation links and adorn them if external
         var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
          for (var i=0; i<links.length; i++) {
            const link = links[i];
            if (!isInternal(link.href)) {
              // undo the damage that might have been done by quarto-nav.js in the case of
              // links that we want to consider external
              if (link.dataset.originalHref !== undefined) {
                link.href = link.dataset.originalHref;
              }
            }
          }
        function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
          const config = {
            allowHTML: true,
            maxWidth: 500,
            delay: 100,
            arrow: false,
            appendTo: function(el) {
                return el.closest('section.slide') || el.parentElement;
            },
            interactive: true,
            interactiveBorder: 10,
            theme: 'light-border',
            placement: 'bottom-start',
          };
          if (contentFn) {
            config.content = contentFn;
          }
          if (onTriggerFn) {
            config.onTrigger = onTriggerFn;
          }
          if (onUntriggerFn) {
            config.onUntrigger = onUntriggerFn;
          }
            config['offset'] = [0,0];
            config['maxWidth'] = 700;
          window.tippy(el, config); 
        }
        const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
        for (var i=0; i<noterefs.length; i++) {
          const ref = noterefs[i];
          tippyHover(ref, function() {
            // use id or data attribute instead here
            let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
            try { href = new URL(href).hash; } catch {}
            const id = href.replace(/^#\/?/, "");
            const note = window.document.getElementById(id);
            if (note) {
              return note.innerHTML;
            } else {
              return "";
            }
          });
        }
        const findCites = (el) => {
          const parentEl = el.parentElement;
          if (parentEl) {
            const cites = parentEl.dataset.cites;
            if (cites) {
              return {
                el,
                cites: cites.split(' ')
              };
            } else {
              return findCites(el.parentElement)
            }
          } else {
            return undefined;
          }
        };
        var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
        for (var i=0; i<bibliorefs.length; i++) {
          const ref = bibliorefs[i];
          const citeInfo = findCites(ref);
          if (citeInfo) {
            tippyHover(citeInfo.el, function() {
              var popup = window.document.createElement('div');
              citeInfo.cites.forEach(function(cite) {
                var citeDiv = window.document.createElement('div');
                citeDiv.classList.add('hanging-indent');
                citeDiv.classList.add('csl-entry');
                var biblioDiv = window.document.getElementById('ref-' + cite);
                if (biblioDiv) {
                  citeDiv.innerHTML = biblioDiv.innerHTML;
                }
                popup.appendChild(citeDiv);
              });
              return popup.innerHTML;
            });
          }
        }
      });
      </script>
    

</body></html>